Arduino/TestCases/TestCases.ino

```diff
@@ -0,0 +1,221 @@
+_data_rate_t dr=DR6;
+_physical_type_t physicalType =EU868 ;
+
+//internal variables
+char data[10];
+char buffer[256];
+int i=0;
+
+
+void setup(void)
+{
+    SerialUSB.begin(115200);
+    while(!SerialUSB);
+    
+    lora.init();  
+    configStandardLoraSettings();
+}
+
+void configLoraOTAA(void)
+{
+    deviceId ="47AAC86800430028";
+    appKey="8AFE71A145B253E49C3031AD068277A1";
+    appEui ="BE7A0000000014E2";   
+    lora.setDeciveMode(LWOTAA);
+    lora.setId(devAddr, deviceId, appEui);
+    lora.setKey(nwkSKey, appSKey, appKey);
+    while(!lora.setOTAAJoin(JOIN,20000));  
+}
+
+void configLoraOTAAWrongDevEUI(void)
+{
+    deviceId ="AAAAC86800430028";
+    appKey="8AFE71A145B253E49C3031AD068277A1";
+    appEui ="BE7A0000000014E2";   
+    lora.setDeciveMode(LWOTAA);
+    lora.setId(devAddr, deviceId, appEui);
+    lora.setKey(nwkSKey, appSKey, appKey);
+    lora.setOTAAJoin(JOIN,10000);  
+    lora.setOTAAJoin(JOIN,10000);
+}
+
+void configLoraABP(void)
+{
+   deviceId ="46AAC86800430028";
+   devAddr ="0028B1B1";
+   appSKey ="2B7E151628AED2A6ABF7158809CF4F3C";
+   nwkSKey ="3B7E151628AED2A6ABF7158809CF4F3C";
+   lora.setDeciveMode(LWABP);
+   lora.setId(devAddr, deviceId, appEui);
+   lora.setKey(nwkSKey, appSKey, appKey);
+}
+
+void configLoraABPWrongDevAddr(void)
+{
+   deviceId ="46AAC86800430028";
+   devAddr ="0028BBBB";
+   appSKey ="2B7E151628AED2A6ABF7158809CF4F3C";
+   nwkSKey ="3B7E151628AED2A6ABF7158809CF4F3C";
+   lora.setDeciveMode(LWABP);
+   lora.setId(devAddr, deviceId, appEui);
+   lora.setKey(nwkSKey, appSKey, appKey);
+}
+
+
+
+void configStandardLoraSettings(void)
+{
+      
+   
+        
+    lora.setDataRate(dr, physicalType);
+    
+    lora.setChannel(0, 868.1);
+    lora.setChannel(1, 868.3);
+    lora.setChannel(2, 868.5);
+    
+    lora.setReceiceWindowFirst(0, 868.1);
+    lora.setReceiceWindowSecond(868.5, DR2);
+    
+    lora.setAdaptiveDataRate(false);
+
+    lora.setDutyCycle(false);
+    lora.setJoinDutyCycle(false);
+    
+    lora.setPower(14);
+}
+
+
+
+void loop(void)
+{   
+    
+  
+    SerialUSB.println("Performing OTAA");       
+    configLoraOTAA();  
+    SerialUSB.println("First or second join should succeed");
+
+    lora.setPort(1);
+    
+    SerialUSB.println("Sending confirmed message, msg should be acknowledged, decoded with DecoderValueSensor and on port=1"); 
+    confirmed=true;
+    sendLoraMessage();
+
+    SerialUSB.println("Sending unconfirmed message"); 
+    confirmed=false;
+    sendLoraMessage();
+
+    SerialUSB.println("Sending 10 confirmed messages, when fcntup=10 it should be saved in the twins"); 
+    confirmed=true;
+    for(int u=0;u<10;u++)
+    {
+      sendLoraMessage();
+    }
+  
+
+    SerialUSB.println("Performing ABP");       
+    configLoraABP();  
+
+    lora.setPort(10);
+
+    SerialUSB.println("Sending confirmed message, msg should be acknowledged, decoded with DecoderValueSensor and port=10"); 
+    confirmed=true;
```

Comment:
Why not have confirmed as function argument? like sendLoRaMessage (bool) ?

Reply:
Ideally the sendLoRaMessage should have multiple parameters like confirmed, msgpayload so we can test size, DR, etc... just did not have enough time to implement it. When we create the automated one we should redesign it in a more customizable way.